podvm/Dockerfile.podvm_builder.rhel

```diff
@@ -8,31 +7,51 @@
 #
 FROM registry.access.redhat.com/ubi9/ubi:9.2
 
+ARG ARCH="amd64"
+ARG PACKER_VERSION="v1.9.4"
 ARG GO_VERSION="1.20.10"
 ARG PROTOC_VERSION="3.11.4"
+ARG PROTOC_ARCH="x86_64"
 ARG RUST_VERSION="1.72.0"
+ARG ORG_ID
+ARG ACTIVATION_KEY
 ARG YQ_VERSION="v4.35.1"
-ARG YQ_CHECKSUM="sha256:bd695a6513f1196aeda17b174a15e9c351843fb1cef5f9be0af170f2dd744f08"
 
-# This must be running on subscribed RHEL host !!!
-# If you are running a UBI container on a registered and subscribed RHEL host, the main RHEL Server repository is enabled inside the standard UBI container
-
-RUN dnf config-manager --set-enabled codeready-builder-for-rhel-9-x86_64-rpms && \
-    dnf groupinstall -y 'Development Tools' && \
-    dnf install -y yum-utils gnupg git perl-core pkg-config libseccomp-devel gpgme-devel \
-        device-mapper-devel qemu-kvm unzip wget libassuan-devel \
-	genisoimage cloud-utils-growpart cloud-init clang tpm2-tss-devel && \
-    yum-config-manager --add-repo https://rpm.releases.hashicorp.com/RHEL/hashicorp.repo && \
-    dnf install -y packer && \
-    dnf clean all && \
-    curl https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz -o go${GO_VERSION}.linux-amd64.tar.gz && \
-    rm -rf /usr/local/go && tar -C /usr/local -xzf go${GO_VERSION}.linux-amd64.tar.gz && \
-    rm -f go${GO_VERSION}.linux-amd64.tar.gz
+ENV SMDEV_CONTAINER_OFF=1
 
-ADD --checksum=${YQ_CHECKSUM} https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64 /usr/local/bin/yq
+# This registering RHEL when building on an unsubscribed system
+# If you are running a UBI container on a registered and subscribed RHEL host, the main RHEL Server repository is enabled inside the standard UBI container
+RUN if [[ -n "${ACTIVATION_KEY}" && -n "${ORG_ID}" ]]; then \
+    subscription-manager register --org=${ORG_ID} --activationkey=${ACTIVATION_KEY}; \
+    fi
+    
+RUN subscription-manager repos --enable codeready-builder-for-rhel-9-${ARCH}-rpms; \
```

Comment:
IIUC codeready expects x86_64 rather than amd64, how about treating the inconsistency using the following:

users will set only the ARCH value as done in other files:
ARG ARCH=x86_64 # set x86_64 or s390x
ENV ARCH=${ARCH}

then:
for protoc use: ${ARCH/s390x/s390_64}
for codeready use: ${ARCH}
for golang use: ${ARCH/x86_64/amd64}
for yq use: ${ARCH/x86_64/amd64}
for packer use:  ${ARCH}